### PR TITLE
JENKINS-39893# Enable findbugs

### PR DESCRIPTION
**Pull request by vivek**
# Description

See [JENKINS-39893](https://issues.jenkins-ci.org/browse/JENKINS-39893).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by vivek** [comment]
findbugs.failOnError is set to true by default. There were few errors, they are fixed so build should go thru fine. From this point on, any new failure will be detected while CI build and must be fixed.

**Comment by vivek** [comment]
@michaelneale PTAL, I have added step in Jenkinsfile to publish fundbugs report.

**Comment by michaelneale** [comment]
rebuilding due to a known problematic test - @cliffmeyers knows what I am talking about.

**Comment by michaelneale** [comment]
ATH doesn't seem happy abotu this - needs investigation.

File: blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/BlueOceanRootAction.java [hunk 1]
```diff
@@ -28,7 +29,12 @@
 @Extension
 public class BlueOceanRootAction implements UnprotectedRootAction, StaplerProxy {
     private static final String URL_BASE="blue";
-    private static final Random randomBits = new Random();
+
+    // Used to indicate UI should send refresh
+    @SuppressFBWarnings(
+            value="DMI_RANDOM_USED_ONLY_ONCE",
+            justification="random long value is used as salt, generated once in Jenkins instance")
+    private static final long randomBits = new Random().nextLong();
```
Comment (vivek) [inline blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/BlueOceanRootAction.java]: Idea behind this is that randomBits is supposed to be a random constant - unique across Jenkins restart. There is sha1 hash calculated down below and set as value of X-Blueocean-Refresher header. Browser does a refresh anytime this value is different - thinking Jenkins restarted. 

The change I made to compute random long for every request was tripping UI to send page refresh on every request, ATH caught it in one of the test.

**Comment by vivek** [comment]
Ugh, build failed, for some reason SupressFBWarning didn't work. looking in to it.

**Comment by michaelneale** [comment]
@vivek if annoying this can wait a week if you have other things to prepare for next week

**Comment by vivek** [comment]
@michaelneale yeah I guess I am going to put it off. Will look at it after next week.

**Comment by michaelneale** [comment]
this could be updated any time, as long as it passes - get it in any time.

**Comment by vivek** [comment]
@michaelneale working on it, I changed threshold from High to Default and there lots of failures. Going thru one by one over the code, should get it in for review once I am done with them.

**Comment by vivek** [comment]
@michaelneale All reported findbugs bugs are fixed and ATH passes https://ci.blueocean.io/blue/organizations/jenkins/ATH-Jenkinsfile/detail/master/2189/pipeline.

**Comment by i386** [comment]
🐝

File: blueocean-dashboard/pom.xml [hunk 2]
```diff
@@ -12,4 +12,16 @@
     <packaging>hpi</packaging>
 
     <url>https://wiki.jenkins-ci.org/display/JENKINS/Blue+Ocean+Plugin</url>
+    <build>
```
Comment (vivek) [inline blueocean-dashboard/pom.xml]: @i386 there is no java code so no point running findbug on this.

Comment (vivek) [inline blueocean-dashboard/pom.xml]: @i386 Reverted. Although in blueocean-dashboard, FB reported bug related to generated Java file for Messages files, I fixed that. That message property is not even used, just had to lower the case to keep FB happy.

**Comment by i386** [comment]
@vivek 🐝

**Comment by vivek** [comment]
ATH passes https://ci.blueocean.io/blue/organizations/jenkins/ATH-Jenkinsfile/detail/master/2200/pipeline.

**Pull request merged**